### ISSUE TITLE
feat[dev]: ✨ implement A11y improvements for Interactive Template

### ISSUE THREAD

**Issue by yoezlem** [solid-design-system/solid#1531]
## Description
This task describes the Storybook implementation for the Interactive template as part of the A11y improvements outlined in Epic https://github.com/solid-design-system/solid/issues/1465.

A11y improvements for the Interactive template can be found here: [Interactive Notes](https://www.notion.so/matuzo/Interactive-112f3f6e1bc08005b5ffce665a1d53e0).

-----------------

## Comment
**Problem:** The button in the icon-only example has no accessible name.
**Solution:** Provide one by labeling the icon.

## Additional Information
- Before working on this template ticket, please finish the main component ticket first. This template ticket is linked to the component and will be resolved once the component's A11y improvements are completed:
#1612 

## DoR
- [x] Item has been estimated by the team
- [x] Item dependencies have been identified and documented

## DoD
- [ ] Documentation has been created/updated (if applicable)
- [ ] Migration Guide has been created/updated (if applicable)
- [ ] Relevant E2E tests (Features, A11y, Bug fixes) are created/updated
- [ ] Relevant stories (Features, A11y) are created/updated
- [ ] Implementation works successfully on `feature` branch
